go-auth.go

```diff
@@ -497,16 +519,31 @@ func setCache(authOpts map[string]string) {
 }
 
 //export AuthUnpwdCheck
-func AuthUnpwdCheck(username, password, clientid string) bool {
+func AuthUnpwdCheck(username, password, clientid string) uint8 {
+	ok, err := authUnpwdCheck(username, password, clientid)
+	if err != nil {
+		log.Error(err)
+		return AuthError
+	}
+
+	if ok {
+		return AuthGranted
+	}
+
+	return AuthRejected
+}
+
+func authUnpwdCheck(username, password, clientid string) (bool, error) {
 	var authenticated bool
 	var cached bool
 	var granted bool
+	var firstError error
```

Comment:
Sorry for bugging you with this, but could we rename it to `err`? There are 3 inner `err` appearances, but 2 are already scoped to their if from their construct, and the other one is being declared at place too (https://github.com/iegomez/mosquitto-go-auth/blob/f1a3fef9ee1aba22682ea5de46551188434431e7/go-auth.go#L601). Maybe that last one could be named `authErr` for clarity if you prefer.